common/src/main/java/com/wynntils/utils/render/FontRenderer.java

```diff
@@ -255,6 +257,183 @@ public void renderAlignedTextInBox(
                 1f);
     }
 
+    public void renderScrollingString(
+            PoseStack poseStack,
+            StyledText styledText,
+            float x,
+            float y,
+            float renderWidth,
+            float translationX,
+            float translationY,
+            CustomColor customColor,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment,
+            TextShadow shadow,
+            float textScale) {
+        int textLength = (int) ((font.width(styledText.getString()) + 1) * textScale);
+
+        if (textLength > renderWidth) {
+            float maxScrollOffset =
+                    switch (horizontalAlignment) {
+                        case CENTER -> -(textLength / 2f) + (renderWidth / 2);
+                        case RIGHT -> 0.0f;
+                        default -> textLength - renderWidth;
+                    };
+
+            double currentTimeInSeconds = (double) Util.getMillis() / 1000.0;
+            double e = Math.max((double) maxScrollOffset * 0.5, 3.0);
+            double f =
+                    Math.sin(1.5707963267948966 * Math.cos(6.283185307179586 * currentTimeInSeconds / e)) / 2.0 + 0.5;
```

Comment:
This feels like Mojang/Mojmap code. Can you extract the big numbers into constants? They are `90 degs` and `360 degs` in radian. Also, can you name the fields to something that makes sense?

Reply:
Yeah was copied from the default `renderScrollingString` missed those two